packages/web-components/fast-foundation/docs/guide/webpack.doc.md

```diff
@@ -3,4 +3,217 @@ id: webpack
+
+Open the `fast-webpack` folder in your favorite editor. You should see your `package.json` along with a `package-lock.json` and a `node_modules` folder.
+
+First, let's create a `src` folder where we'll put all our TypeScript code. In the `src` folder, add a `main.ts` file. You can leave the file empty for now. We'll come back it in a bit.
+
+Next, in the root of your project folder, add a `tsconfig.json` file to configure the TypeScript compiler. Here's an example starter config that you can put into that file:
+
+```json
+{
+  "compilerOptions": {
+    "pretty": true,
+    "target": "ES2015",
+    "module": "ES2015",
+    "moduleResolution": "node",
+    "importHelpers": true,
+    "experimentalDecorators": true,
+    "declaration": true,
+    "declarationMap": true,
+    "sourceMap": true,
+    "noEmitOnError": true,
+    "strict": true,
+    "outDir": "dist/build",
+    "rootDir": "src",
+    "lib": [
+      "dom",
+      "esnext"
+    ]
+  },
+  "include": [
+    "src"
+  ]
+}
+```
+
+You can learn more about `tsconfig.json` options in [the official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html).
+
+Next, create a `webpack.config.js` file in the root of your project folder with the following source:
+
+```js
+module.exports = function(env, { mode }) {
+  const production = mode === 'production';
+  return {
+    mode: production ? 'production' : 'development',
+    devtool: production ? 'source-maps' : 'inline-source-map',
+    entry: {
+      app: ['./src/main.ts']
+    },
+    output: {
+      filename: 'bundle.js'
+    },
+    resolve: {
+      extensions: ['.ts', '.js'],
+      modules: ['src', 'node_modules']
+    },
+    devServer: {
+      port: 9000,
+      historyApiFallback: true,
+      writeToDisk: true,
+      open: !process.env.CI,
+      lazy: false
+    },
+    module: {
+      rules: [
+        {
+          test: /\.ts$/i,
+          use: [
+            {
+              loader: 'ts-loader'
+            }
+          ],
+          exclude: /node_modules/
+        }
+      ]
+    }
+  }
+}
+```
+
+This setup uses `ts-loader` to process TypeScript. It will also enable both a production mode and a development mode that watches your source, recompiling and refreshing your browser as things change. You can read more about Webpack configuration in [the official Webpack documentation](https://webpack.js.org/).
+
+To enable easy execution of both our production and development builds, let's add some script commands to our `package.json` file. Find the `scripts` section of your `package.json` file and add the following two scripts:
+
+```json
+"scripts": {
+  "build": "rimraf dist && webpack --config webpack.config.js --mode=production",
```

Comment:
weblack has a "clean" plugin - maybe we should reference that instead of `rimraf` to keep the tool-chain consistent?

https://webpack.js.org/guides/output-management/#cleaning-up-the-dist-folder

Reply:
Will explore this evening and see if we can go that way. Thanks for the tip!

Reply:
We're using the clean plugin in the `fast-website` project.

`npm install --save-dev clean-webpack-plugin`

then in webpack.config.js: 
```
const { CleanWebpackPlugin } = require("clean-webpack-plugin");

module.exports = {
    plugins: [
        new CleanWebpackPlugin(),
...
```